src/dynamite/_backend/bsubspace_impl.h

```diff
@@ -61,6 +61,10 @@ static inline PetscInt S2I_nocheck_Full(PetscInt state, const data_Full* data) {
 }
 
 static inline PetscInt I2S_Full(PetscInt idx, const data_Full* data) {
+  PetscInt ierr;
+  PetscAssert(idx >= 0 && idx < Dim_Full(data), PETSC_COMM_WORLD, ierr, 
```

Comment:
We should change PETSC_COMM_WORLD to PETSC_COMM_SELF everywhere

Reply:
changed